### PR TITLE
Minimum WordPress and WooCommerce versions

### LINKED ISSUE #799: Minimum supported WP and WC versions
### Describe the bug:
We currently have the minimum supported versions set to:

```
Requires at least: 5.3
WC requires at least: 4.5
```

Although the Jetpack connection package we use, calls the function `wp_get_environment_type`, which was introduced in WordPress 5.5. In older versions it used to check if this function exists, but since we [updated the Jetpack connect package](https://github.com/woocommerce/google-listings-and-ads/pull/336), that's no longer the case.

So this causes a fatal error on any site installing on a version older then 5.5. Do we need to bump any of these versions or create some compatibility if we want to support older versions?

### Steps to reproduce:
1. Setup a site on WordPress 5.4 and an older version of WooCommerce
2. Install and activate the GLA extension
3. Observe the fatal error

### Additional details:
Originally reported in https://wordpress.org/support/topic/fatal-error-3893/

### PR DESCRIPTION
**Pull request by mikkamp**
### Changes proposed in this Pull Request:

This PR bumps the minimum WordPress and WooCommerce versions to 2 versions less then latest, based on [L2 support policy](https://developer.woocommerce.com/what-we-support/)

It also adds a WooCommerce version validator (after plugins are loaded).

Closes #799

### Screenshots:
image

### Detailed test instructions:

1. Activate the plugin on a site with an older version of WordPress (< 5.5)
2. Observe that WordPress gives a warning "Error: Current WordPress version does not meet minimum requirements for Google Listings and Ads."
3. Activate the plugin on a site with an older version of WooCommerce (< 5.2)
2. Observe the admin notice warning the minimum version, none of the GLA pages are available in the menu

### Changelog entry
* Update - Increase required minimum WordPress and WooCommerce versions.

### CODE REVIEW

File: src/Internal/Requirements/VersionValidator.php [hunk 1]
```diff
@@ -36,7 +37,19 @@ public function validate(): bool {
...
+	}
+
+	/**
+	 * Validate WordPress version being used.
+	 *
+	 * @throws InvalidVersion When the WordPress version does not meet the minimum version.
+	 */
+	protected function validate_wp_version() {
+		$wp_version = get_bloginfo( 'version' );
+		if ( ! version_compare( $wp_version, '5.5', '>=' ) ) {
```
Comment (nima-karimi) [inline src/Internal/Requirements/VersionValidator.php]: Can we use constants for all these minimum versions? I think it'd be easier to change later if we have them all defined in one place (possibly in the  `google-listings-and-ads.php` file where the minimum versions are  commented)

**Comment by nima-karimi** [comment]
I tested this using WP 5.4.6 and WC 5.1 and got the following notice after activating the GLA plugin:

image

This was displayed on a new page (and not as an admin notice). WP shows this error when the version doesn't match the minimum defined in the plugin header. Do we also need to add another notice for WP? Not sure if it's even displayed.

**Comment by mikkamp** [comment]
> WP shows this error when the version doesn't match the minimum defined in the plugin header. Do we also need to add another notice for WP? Not sure if it's even displayed.

You can get around that if you install an already released version of GLA and activate it, and then update it outside of the regular WP dashboard. Once we release it to WP.org it will show a warning that it's not compatible when updating.

So it can be a bit overkill since there will be few sites that update like this. Happy to revert the WP check if we feel that it's a bit unnecessary.

Edit: Had a look at AW and there they only warn for the upcoming minimum WP version, so we should be alright to remove this requirement check.

**Comment by nima-karimi** [comment]
Since this check is already built-in WP we can probably remove our own check and rely on that. I guess few people will upgrade that way without going through the WP dashboard. Especially since it prevents them from installing it.

**Comment by nima-karimi** [comment]
Ah, a warning would be nice. We can add it later though.

**Comment by mikkamp** [comment]
Removed the WP check and added the constants. It's ready for another round of review.

**Review by nima-karimi — APPROVED** [review]
Thanks for making the changes 💯 I get the admin notice when using WC 5.1:

image

**Pull request merged**